From 248d49873cd968221cb86781e17a06a0ad587d8d Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexdeucher@gmail.com>
Date: Fri, 20 May 2011 04:34:30 -0400
Patch-mainline: 3.0-rc3
References: fate#310916
Git-commit: 7c3ed0fd9b72d489243ff97f3cea2e1d4d07d9fa
Subject: [PATCH 2466/2588] drm/radeon/kms: bail early for eDP in hotplug
 callback

Don't try and en/disable the port as it may be a hpd event from
powering up/down the panel during a modeset or dpms.

Signed-off-by: Alex Deucher <alexdeucher@gmail.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 7c3ed0fd9b72d489243ff97f3cea2e1d4d07d9fa)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/radeon_connectors.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_connectors.c b/drivers/gpu/drm/radeon/radeon_connectors.c
index 5947871..532f17d 100644
--- a/drivers/gpu/drm/radeon/radeon_connectors.c
+++ b/drivers/gpu/drm/radeon/radeon_connectors.c
@@ -52,6 +52,12 @@ void radeon_connector_hotplug(struct drm_connector *connector)
 
 	radeon_hpd_set_polarity(rdev, radeon_connector->hpd.hpd);
 
+	/* powering up/down the eDP panel generates hpd events which
+	 * can interfere with modesetting.
+	 */
+	if (connector->connector_type == DRM_MODE_CONNECTOR_eDP)
+		return;
+
 	/* pre-r600 did not always have the hpd pins mapped accurately to connectors */
 	if (rdev->family >= CHIP_R600) {
 		if (radeon_hpd_sense(rdev, radeon_connector->hpd.hpd))
-- 
1.7.6

